### ISSUE TITLE
"User Profile" attributes not available for Users Attribute search and Attribute selection, if no view or manage realm realm-management role added

### ISSUE THREAD

**Issue by mpalavrov** [keycloak/keycloak#27536]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

account/ui

### Describe the bug

Realm Admin users, having "realm-management" role/permissions only to "manage-users" and "query-users", cannot make Search using the Attribute or to add attribute for example for Locale.
Having "view-realm" or of course "manage-realm" is fixing it, but then the Realm Admin users are also able to see the Realm Settings, which might be not desired as it contains sensitive information.

Please note that the "User Profile" attributes are available normally, as the User Profile "groups" and "attributes" are localized when the User is opened

### Version

24.0.1

### Regression

- [x] The issue is a regression

### Expected behavior

The User Profile Attributes are available for User Attribute search without the need of "view-realm" or "manage-realm" realm-management Client Role to be assigned:
User attribute search:
![image](https://github.com/keycloak/keycloak/assets/33352511/cbc62e3a-a10e-43be-8b24-bc6a28065504)
Locale attribute selection:
![image](https://github.com/keycloak/keycloak/assets/33352511/658f8784-0cba-439b-a2ed-d81fb1f47e1c)

### Actual behavior

The User Attribute Search is not usable, as User Profile attributes are not available in case "view-realm" or "manage-realm" realm-management Client Role is not assigned:
User attribute search:
![image](https://github.com/keycloak/keycloak/assets/33352511/d44130bf-d778-4d5e-956c-14bbc3960ccf)
User Attribute select:
![image](https://github.com/keycloak/keycloak/assets/33352511/87f88219-7c9a-4abb-a9d6-fbdd8f27507b)

### How to Reproduce?

1. Create User Profile attributes, also some with multiselect, like for Locale selection
2. Create user with user/credentials (or use Identity Provider if available)
3. Assign "manage-users" and "query-users" realm-management Client roles to an user
4. Sign in with the security-admin-console client with 
5. Go to the User menu:
![image](https://github.com/keycloak/keycloak/assets/33352511/25fd6e91-6d6c-41b7-b027-3f816d5c5f60)
6. Try to make Attribute Search:
![image](https://github.com/keycloak/keycloak/assets/33352511/84ea5c3b-be3b-4d1b-986e-66abd9768a19)
or to assign some Attribute for the user

**Comment by enastevska** [comment]
Is there an update when this will be resolved?

**Comment by mpalavrov** [comment]
Hey @ssilvert, could you please provide some update on this bug? It is still there for KC 25.0.1.
Thank you.

**Comment by antikalk** [comment]
I just verified that the issue still persists with KC 25.0.2.

**Comment by danielFesenmeyer** [comment]
@mpalavrov checked this with KC 24.0.6. Seems like the attribute selection has been fixed, I was able to select and save an "options-attribute" with "manage-users" role.

But the User Attribute Search issue still persists. Will also check on main and see whether I can provide a PR.

**Comment by danielFesenmeyer** [comment]
Hi @ssilvert,

I've implemented a draft, which shows how this issue can be solved from my POV: https://github.com/keycloak/keycloak/pull/31771

Actually, it's still a draft because I've added a new "ui-ext" endpoint on realm level, which allows access to all users having any admin rights. This new endpoint is now used to retrieve the information whether userProfileProviders are enabled, but could also be used for similar use cases. Before, the corresponding "component" was called, which requires at least "view-realm" permission, breaking the users view in case the logged-in admin only has "view-users" rights.

I've added several tests for the functionality and think the PR is at least a good starting point for the final fix.
